Mathlib/Analysis/NormedSpace/Banach.lean

```diff
@@ -545,3 +545,43 @@ theorem coe_ofSeqClosedGraph
 end ContinuousLinearMap
 
 end ClosedGraphThm
+
+section BijectivityCriteria
+
+namespace ContinuousLinearMap
+
+variable [CompleteSpace E]
+
+lemma closed_range_of_antilipschitz {f : E →SL[σ] F} {c : ℝ≥0} (hf : AntilipschitzWith c f) :
+    (LinearMap.range f).topologicalClosure = LinearMap.range f :=
+  SetLike.ext'_iff.mpr <| (hf.isClosed_range f.uniformContinuous).closure_eq
+
+open Function
+lemma bijective_iff_dense_range_and_antilipschitz (f : E →SL[σ] F) :
+    Bijective f ↔ (LinearMap.range f).topologicalClosure = ⊤ ∧ ∃ c, AntilipschitzWith c f := by
+  refine ⟨fun h ↦ ⟨?eq_top, ?anti⟩, fun ⟨hd, c, hf⟩ ↦ ⟨hf.injective, ?surj⟩⟩
+  case eq_top => simpa [SetLike.ext'_iff] using h.2.denseRange.closure_eq
+  case anti =>
+    have := ContinuousLinearEquiv.ofBijective f ?_ ?_ |>.antilipschitz
+    · exact ⟨_, by simpa⟩
+    all_goals simp only [LinearMap.range_eq_top, LinearMapClass.ker_eq_bot]
+    exacts [h.1, h.2]
+  case surj => rwa [← LinearMap.range_eq_top, ← closed_range_of_antilipschitz hf]
+
+lemma _root_.AntilipschitzWith.completeSpace_range_clm {f : E →SL[σ] F} {c : ℝ≥0}
+    (hf : AntilipschitzWith c f) : CompleteSpace (LinearMap.range f) :=
+  IsClosed.completeSpace_coe <| hf.isClosed_range f.uniformContinuous
+
+lemma isUnit_iff_bijective {f : E →L[𝕜] E} : IsUnit f ↔ Bijective f := by
```

Comment:
I feel like this should be a bit earlier in the file? I would say newt to `ContinuousLinearEquiv.ofBijective`